corehq/apps/app_manager/models.py

```diff
@@ -4076,6 +4092,11 @@ def wrap(cls, data):
                 data['text_input'] = 'native' if data['native_input'] else 'roman'
             del data['native_input']
 
+        if 'build_langs' in data:
+            if (data['build_langs'] != data['langs']) and ('build_profiles' not in data):
+                    data['build_profiles'] = {uuid.uuid4().hex : BuildProfile(name=', '.join(data['build_langs']), langs=data['build_langs'])}
```

Comment:
I wonder if this could be buggy, coming up with a new uuid like this on the fly. It allows for an edge case in which the app is wrapped `build_langs` is migrated in-memory but not saved, one of the new uuids is used somewhere (like sent over to an html page), and then on a subsequent request that unsaved uuid is used, but doesn't match any of the uuids in the newly wrapped doc. I'd either find some way around that edge case (whether or not it's actually exposed), like using some sort of predictable hashing to generate the key or making sure it gets saved when it's migrated, or else add a comment here in the code about why you're pretty sure this edge case is never exposed. (That'll give someone who's hunting down a bug at least a place to start from to decide whether or not it's related.)